src/smc-webapp/course/main.cjsx

```diff
@@ -1223,7 +1225,188 @@ exports.init_redux = init_redux = (redux, course_project_id, course_filename) ->
             # Now open it
             redux.getProjectActions(proj).open_directory(path)
 
+        # Handouts
```

Comment:
Simplified copies of much of what's happening in similarly named assignment methods. Some cases you could literally change `handout` for `assignment` and you'd have the other function. Keeping separate same as previous changes.